src/projectscene/internal/projectviewstate.cpp

```diff
@@ -19,6 +19,95 @@ void ProjectViewState::changeTracksVericalY(int deltaY)
     m_tracksVericalY.set(deltaY);
 }
 
+qreal ProjectViewState::mousePositionY() const
+{
+    return m_mouseYPosition.val;
+}
+
+void ProjectViewState::setMousePositionY(qreal y)
+{
+    m_mouseYPosition.set(y);
+}
+
+au::trackedit::TrackId ProjectViewState::determinePointedTrack(double y) const
+{
+    trackedit::ITrackeditProjectPtr prj = globalContext()->currentTrackeditProject();
+    if (!prj) {
+        return {};
+    }
+
+    if (y < 1) {
+        y = 1;
+    }
+
+    trackedit::TrackIdList tracks = prj->trackIdList();
+
+    int tracksVericalY = this->tracksVericalY().val;
+    int trackTop = -tracksVericalY;
+    int trackBottom = trackTop;
+
+    for (trackedit::TrackId trackId : tracks) {
+        trackTop = trackBottom;
+        trackBottom = trackTop + trackHeight(trackId).val;
+
+        if (y > trackTop && y < trackBottom) {
+            return trackId;
+        }
+    }
+
+    return -1;
```

Comment:
let's add a constant next to the definition of TrackId something like INVALID_TRAKID = -1;